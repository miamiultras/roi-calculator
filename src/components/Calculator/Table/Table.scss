.table-container {
  max-width: 650px;
  margin: 0 auto;
  display: flex;
  justify-content: center;

  table, td, th {
    font-size: 18px;
    font-weight: 300;
    border-bottom: 2px solid #ddd;
    text-align: center;
  }

  table {
    border-collapse: collapse;
    width: 100%;
  }

  th, td {
    padding: 20px;
  }

  tr td:first-child {
    text-align: left;
  }

  img {
    width: 130px;
    height: auto;
  }
}

@media only screen and (max-width: 540px) {
    .table-container {
      table, td, th {
        font-size: 16px;
        font-weight: 300;
        border-bottom: 2px solid #ddd;
        text-align: center;
      }
      th, td {
        padding: 10px;
      }
    }
}

@media only screen and (max-width: 382px) {
  .table-container {
    table, td, th {
      font-size: 14px;
      font-weight: 300;
      border-bottom: 2px solid #ddd;
      text-align: center;
    }

    th, td {
      padding: 8px;
    }
  }
}
