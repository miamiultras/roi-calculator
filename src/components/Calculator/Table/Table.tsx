import React from 'react';
import './Table.scss';
import { Service } from '../../../interfaces/service.interface';

const Table = ({ data }: { data: Service[] }) => (
  <div className='table-container'>
    <table>
      <thead>
        <tr>
          <th />
          <th>Home Grown</th>
          <th>
            <img src='https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7' alt='Cleeng' />
          </th>
        </tr>
      </thead>
      <tbody>
        {data.map((service: Service) => (
          <tr key={service.name + Math.random()}>
            <td>{service.name}</td>
            <td>{service.homeGrown}</td>
            <td>{service.cleeng}</td>
          </tr>))
        }
      </tbody>
    </table>
  </div>
);

export default Table;
