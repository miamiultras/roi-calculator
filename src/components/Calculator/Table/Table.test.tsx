import * as React from 'react';
import { shallow } from 'enzyme';
import Table from './Table';

function setup() {
  const props = {};
  const wrapper = shallow(<Table data={[]} />);
  return {
    wrapper,
    props,
  };
}

describe('Table component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
