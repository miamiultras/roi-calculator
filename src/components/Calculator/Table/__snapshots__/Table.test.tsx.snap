// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    data={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table>
        <thead>
          <tr>
            <th />
            <th>
              Home Grown
            </th>
            <th>
              <img
                alt="Cleeng"
                src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
              />
            </th>
          </tr>
        </thead>
        <tbody />
      </table>,
      "className": "table-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th />
              <th>
                Home Grown
              </th>
              <th>
                <img
                  alt="Cleeng"
                  src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                />
              </th>
            </tr>
          </thead>,
          <tbody />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th />
              <th>
                Home Grown
              </th>
              <th>
                <img
                  alt="Cleeng"
                  src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                />
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th />,
                <th>
                  Home Grown
                </th>,
                <th>
                  <img
                    alt="Cleeng"
                    src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                  />
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home Grown",
                },
                "ref": null,
                "rendered": "Home Grown",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Cleeng"
                    src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Cleeng",
                    "src": "https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table>
          <thead>
            <tr>
              <th />
              <th>
                Home Grown
              </th>
              <th>
                <img
                  alt="Cleeng"
                  src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                />
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        "className": "table-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th />
                <th>
                  Home Grown
                </th>
                <th>
                  <img
                    alt="Cleeng"
                    src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                  />
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th />
                <th>
                  Home Grown
                </th>
                <th>
                  <img
                    alt="Cleeng"
                    src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                  />
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th />,
                  <th>
                    Home Grown
                  </th>,
                  <th>
                    <img
                      alt="Cleeng"
                      src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                    />
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home Grown",
                  },
                  "ref": null,
                  "rendered": "Home Grown",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Cleeng"
                      src="https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Cleeng",
                      "src": "https://cdn-statics.cleeng.com/branding/images/logo-small.png?v=db9a54bbc7",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
